src/evict/evict_lru.c

```diff
@@ -914,6 +917,17 @@ __wt_evict_file_exclusive_on(WT_SESSION_IMPL *session)
     if (F_ISSET(session, WT_SESSION_PREFETCH_ENABLED))
         WT_ERR(__wt_conn_prefetch_clear_tree(session, false));
 
+    /*
+     * To avoid a race condition, check if this dhandle is currently being operated on by any other
+     * prefetch thread. If so, wait for the thread to finish before proceeding with eviction.
+     */
+    while (i < WT_PREFETCH_THREAD_COUNT) {
+        if (conn->prefetch_dhandles[i] == session->dhandle) {
```

Comment:
And I think this needs to be a `WT_ORDERED_READ`, too.

Reply:
I assume we want the barrier to ensure a fresh read of the variable on each loop iteration? I don't see any ordering between two variables that'd be required.

Reply:
Good question. If it was just to ensure a fresh read, we could use `WT_READ_ONCE`, and we definitely need at least that.

My slightly hand-wavy continued response is that a thread reading this variable, and another thread writing it, without any form of barrier, is a data race regardless of what we end up using that variable for, or what the requirements of the surrounding code are.

Reply:
I've changed this to `WT_READ_ONCE` [ae383e8](https://github.com/wiredtiger/wiredtiger/pull/10189/commits/ae383e82604d374efa961601fb9e9406cf492771)

Reply:
I agree `READ_ONCE` semantics are what we need here, but for a handful of reasons we don't have `WRITE_ONCE` in the code base and so we can't pair the `READ_ONCE`s with a `WRITE_ONCE` on the writer side. 

I'd recommend we use `WT_PUBLISH` and `WT_ORDERED_READ` instead and add comments that say we're only using these macros for the read/write_once semantics and don't need the ordering constraints they also deliver.

Reply:
Done [8044dc8](https://github.com/wiredtiger/wiredtiger/pull/10189/commits/8044dc8a13a4f60dbf15faf7565d1e28ca8afdac)